Unable to build Tarantool on MacOS

```
-- Performing Test MACOS_V1012 - Success
-- Looking for gethostbyname in nsl
[ 32%] Built target module_api
[ 32%] Generating sql/opcodes.h
[ 32%] Linking C executable uri.test
-- Looking for gethostbyname in nsl - not found
-- Looking for gethostbyname in socket
[ 32%] Built target uri.test
In file included from /Users/o.babin/Projects/tarantool/test/unit/bloom.cc:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/c++/v1/unordered_set:390:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/c++/v1/__hash_table:15:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/c++/v1/memory:2666:26: error: unexpected type name '_CompressedPair': expected expression
    struct _ALIGNAS_TYPE(_CompressedPair) _Storage {
                         ^
/Users/o.babin/Projects/tarantool/src/lib/core/crash.c:400:14: error: 'vfork' is deprecated: Use posix_spawn or fork [-Werror,-Wdeprecated-declarations]
        pid_t pid = vfork();
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/unistd.h:604:1: note: 'vfork' has been explicitly marked deprecated here
__deprecated_msg("Use posix_spawn or fork")
^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/sys/cdefs.h:208:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
1 error generated.
make[2]: *** [src/lib/core/CMakeFiles/core.dir/crash.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
-- Looking for gethostbyname in socket - not found
-- Looking for socket in socket
[ 32%] Linking CXX executable bps_tree_iterator.test
/Users/o.babin/Projects/tarantool/src/lib/core/popen.c:1184:16: error: 'vfork' is deprecated: Use posix_spawn or fork [-Werror,-Wdeprecated-declarations]
        handle->pid = vfork();
                      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/unistd.h:604:1: note: 'vfork' has been explicitly marked deprecated here
__deprecated_msg("Use posix_spawn or fork")
^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/sys/cdefs.h:208:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
1 error generated.
make[2]: *** [src/lib/core/CMakeFiles/core.dir/popen.c.o] Error 1
```

--- a/src/lib/core/crash.c
+++ b/src/lib/core/crash.c
@@ -9,6 +9,7 @@
 #include <stdint.h>
 #include <limits.h>
 #include <time.h>
+#include <spawn.h>
 #include <sys/types.h>
 #include <sys/utsname.h>
 
@@ -392,29 +393,13 @@ crash_report_feedback_daemon(struct crash_info *cinfo)
 		[6] = NULL,
 	};
 
-	/*
-	 * Can't use fork here because libev has own
-	 * at_fork helpers with mutex where we might
-	 * stuck (see popen code).
-	 */
-	pid_t pid = vfork();
-	if (pid == 0) {
-		extern char **environ;
-		/*
-		 * The script must exit at the end but there
-		 * is no simple way to make sure from inside
-		 * of a signal crash handler. So just hope it
-		 * is running fine.
-		 */
-		execve(exec_argv[0], exec_argv, environ);
-		pr_crit("exec(%s,[%s,%s,%s,%s,%s,%s,%s]) failed",
-			exec_argv[0], exec_argv[0],
-			exec_argv[1], exec_argv[2],
-			exec_argv[3], exec_argv[4],
-			exec_argv[5], exec_argv[6]);
-		_exit(1);
-	} else if (pid < 0) {
-		pr_crit("unable to vfork (errno %d)", errno);
+	extern char **environ;
+	int rc = posix_spawn(NULL, exec_argv[0], NULL, NULL, exec_argv, environ);
+	if (rc != 0) {
+		pr_crit("posix_spawn with "
+			"exec(%s,[%s,%s,%s,%s,%s,%s,%s]) failed: %s", exec_argv[0],
+			exec_argv[0], exec_argv[1], exec_argv[2], exec_argv[3],
+			exec_argv[4], exec_argv[5], exec_argv[6], strerror(rc));
 		return -1;
 	}
 

--- a/src/lib/core/popen.c
+++ b/src/lib/core/popen.c
@@ -1178,7 +1178,15 @@ popen_new(struct popen_opts *opts)
 	 * vfork to complete. Also we try to do as minimum
 	 * operations before the exec() as possible.
 	 */
+#pragma GCC diagnostic push
+	/*
+	 * TODO(gh-6674): we need to review popen's design and refactor this
+	 * part, completely getting rid of vfork, or change vfork to
+	 * posix_spawn here, but only on macOS.
+	 */
+#pragma GCC diagnostic warning "-Wdeprecated-declarations"
 	handle->pid = vfork();
+#pragma GCC diagnostic pop
 	if (handle->pid < 0) {
 		diag_set(SystemError, "vfork() fails");
 		goto out_err;

